core/DataTable.php

```diff
@@ -1233,6 +1236,30 @@ public function getSerialized($maximumRowsInDataTable = null,
         return $aSerializedDataTable;
     }
 
+    private static $previousRowClasses = array('O:39:"Piwik\DataTable\Row\DataTableSummaryRow"', 'O:19:"Piwik\DataTable\Row"', 'O:36:"Piwik_DataTable_Row_DataTableSummary"', 'O:19:"Piwik_DataTable_Row"');
+    private static $rowClassToUseForUnserialize   = 'O:29:"Piwik_DataTable_SerializedRow"';
+
+    /**
+     * We cannot unserialize directly as it won't work to unserialize eg "Piwik\DataTable\Row" directly on
+     * PHP 5.6, 5.4.29 and 5.5.13. The Row class implements the "Serializable" interface and can therefore not be
+     * unserialized directly. This would trigger a notice and fail. Instead we use a class "Piwik_DataTable_SerializedRow"
+     * to unserialize which does not implement anything just to be able to access its "$row->c" property later.
+     * @param string $serialized
+     * @return array
+     * @throws Exception In case the unserialize fails
+     */
+    private function unserializeRows($serialized)
```

Comment:
`ArrayObject` implements the `Serializable` interface therefore we cannot simple unserialize the serialized `Row` instances see comment. This was the tricky part in making it work with `ArrayObject`